localstack/aws/app.py

```diff
@@ -31,6 +32,7 @@ def __init__(self, service_manager: ServiceManager = None) -> None:
                 handlers.add_internal_request_params,
                 handlers.handle_runtime_shutdown,
                 metric_collector.create_metric_handler_item,
+                load_service_for_data_plane,
```

Comment:
more work for the handler chain 🥲

Reply:
this is a very cheap operation and it's not doing duplicate work. if this already determines the service, then the service loader does no extra work.